### ISSUE TITLE
Private Connection data sources not available in v0.6.0

### ISSUE THREAD

**Issue by joelsdc** [elastic/terraform-provider-ec#595]
## Readiness Checklist

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed
- [X] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
The `ec_gcp_private_service_connect_endpoint` datasource to work

## Current Behavior
```
│ Error: Invalid data source
│
│   on service_elastic.tf line 7, in data "ec_gcp_private_service_connect_endpoint" "default":
│    7: data "ec_gcp_private_service_connect_endpoint" "default" {
│
│ The provider elastic/ec does not support data source "ec_gcp_private_service_connect_endpoint".
```

## Terraform definition:

As per documentation:

```hcl
data "ec_gcp_private_service_connect_endpoint" "default" {
  region = "us-central1"
}
```

## Steps to Reproduce
1. Add a `ec_gcp_private_service_connect_endpoint` datasource
2. Try to run terraform plan/apply

## Context
After upgrading to v0.6.0 of the provider the `ec_gcp_private_service_connect_endpoint` datasource no longer works.
On v0.5.1 `ec_gcp_private_service_connect_endpoint` works correctly.

## Your Environment
* Version used: v0.6.0
* Terraform cli: v1.3.9

**Comment by joelsdc** [comment]
Looking at the diff between v0.5.1 and v0.6.0 I have the feeling that **all datasources** are broken...

**Comment by joelsdc** [comment]
@tobio tagging you.

**Comment by tobio** [comment]
@joelsdc sorry about the frustration here. tl;dr I'd stick with 0.5.1 until there's a follow up release here. 

It's expected that the `ec_gcp_private_service_connect_endpoint`, `ec_aws_privatelink_endpoint`, and `ec_azure_privatelink_endpoint` data sources aren't available in 0.6. The release notes indicate that changes made in 0.5x are not expected to be part of 0.6, however we can definitely be more explicit about what's missing. I'll chase this up. 

At some point during the development of the migration to the TF plugin framework (i.e the 0.6.0 release) it was decided to perform a wholesale migration based on the 0.4 branch, and follow up with the 0.5 changes in a later release. The result is a pretty frustrating user experience, I understand this is far from ideal for you right now. 

@dimuon is there work underway to migrate the private link data sources (along with the other 0.5x changes) in a 0.6.1 release? Do we have a timeframe here? 

> I have the feeling that all datasources are broken

@dimuon are you able to take a look into this. FWIW the `ec_stack` is working for me in 0.6.0

**Comment by joelsdc** [comment]
Thanks @tobio for checking this, I'm editing my comment from above, I was referring to all **private-connection** data sources (meaning GCP, AWS and Azure). I myself should have been more specific.

My recommendation, until the data sources are ported to the 0.6 branch, would be to:

1- Remove them from the docs: https://registry.terraform.io/providers/elastic/ec/latest/docs

<img width="1068" alt="image" src="https://user-images.githubusercontent.com/16212586/222326024-c9473d29-f007-4e00-958a-984a036a85fb.png">

2- Make it crystal clear in the **README** / **CHANGELOG** / **etc** that they are not available.

I know it's mentioned in the changelog that "0.6 is based on 0.4 and changes added in 0.5 are not available in 0.6", so that leaves it to the user of the provider to go figure out what was added in 0.5 to know it's not available in 0.6, but then the 0.6 documentation says otherwise, which one is the source of truth?

I'm going to stick to 0.5 until this is solved so stay tuned for more bug reports ;)
